### PR TITLE
Receive button properties from the Checkout Block (3202)

### PR DESCRIPTION
**Pull request by danieldudzic**
This is some exploratory work to explore how we can synchronise the express payment button styles better to create more cohesiveness in the Express Checkout block. This PR is a part of https://github.com/woocommerce/woocommerce/pull/47899. It updates the express payment method to receive `buttonAttributes` property from the checkout block, and override the `height` property if available.

### Testing instructions

- Run woocommerce locally and check out the `try/poc-express-checkout-button-styles` branch
- Make sure the environment is right for one of the express checkout buttons to be visible (either Apple or Google Pay)
- Add some items to the cart and visit the cart page (with the cart block)
- Ensure the express payment method is visible there
- Click "Proceed to checkout"
- Ensure the express payment method is visible there too
- In a new tab, open the checkout page in the editor.
- Click on the express checkout area and change the button size to "Small"
- Refresh the checkout page in the front-end
- You should notice the button change size to a height of 40px

Please note that is not possible to set "Large" size as PayPal only allows values between [25px and 55px](https://developer.paypal.com/docs/multiparty/checkout/standard/customize/buttons-style-guide/#size):
```
Error: Expected style.height to be between 25px and 55px - got 56px
```

### Comment from @Dinamiko 

Hi @alexflorisca, just to let you know that we have created this PR for working on https://github.com/woocommerce/woocommerce/pull/47899 POC integration, for now it only checks if `buttonAttributes` exist and if so it gets the button height value from there.

I noticed that is not currently possible to set "Large" size because there is a button size limitation on PayPal.

cc. @InpsydeNiklas

### CODE REVIEW

**Comment by alexflorisca** [comment]
> Please note that is not possible to set "Large" size as PayPal only allows values between [25px and 55px](https://developer.paypal.com/docs/multiparty/checkout/standard/customize/buttons-style-guide/#size):

This has now been fixed on the branch in Woo `try/poc-express-checkout-button-styles`, so you should get 55px for Large buttons.

File: modules/ppcp-blocks/resources/js/checkout-block.js [hunk 1]
```diff
@@ -653,19 +664,47 @@ const PayPalComponent = ( {
...
+		if ( buttonAttributes ) {
+			return {
+				...configStyle,
+				height: buttonAttributes.height
+					? Number( buttonAttributes.height )
+					: configStyle.height,
+				borderRadius: buttonAttributes.borderRadius
+					? Number( buttonAttributes.borderRadius )
+					: configStyle.borderRadius,
+				color: buttonAttributes.darkMode ? 'white' : configStyle.color,
```
Comment (alexflorisca) [inline modules/ppcp-blocks/resources/js/checkout-block.js]: We've removed `darkMode` prop in https://github.com/woocommerce/woocommerce/pull/51109. Please remove it from here too.

File: modules/ppcp-blocks/resources/js/checkout-block.js [hunk 2]
```diff
@@ -582,6 +583,16 @@ const PayPalComponent = ( {
...
 	);
 
+	if ( typeof buttonAttributes !== 'undefined' ) {
+		style.height = buttonAttributes?.height
+			? Number( buttonAttributes.height )
+			: style.height;
+		style.borderRadius = buttonAttributes?.borderRadius
+			? Number( buttonAttributes.borderRadius )
+			: style.borderRadius;
+		style.color = buttonAttributes?.darkMode ? 'white' : '';
```
Comment (alexflorisca) [inline modules/ppcp-blocks/resources/js/checkout-block.js]: We've removed `darkMode` prop in https://github.com/woocommerce/woocommerce/pull/51109. Please remove it from here too.

**Review by Longanh89 — APPROVED** [review]

**Review by alexflorisca — APPROVED** [review]
So good to see this come to light, thanks for doing the work to implement it. Tests well and code looks good to me 👍

**Pull request merged**